pkg/alerting/alert_handler.go

```diff
@@ -6,24 +6,54 @@ import (
 	"net/http"
 	"time"
 
+	"go.uber.org/fx"
+	"go.uber.org/zap"
+	"golang.org/x/exp/maps"
+
 	"github.com/uptrace/bun"
 	"github.com/uptrace/bunrouter"
+	"github.com/uptrace/go-clickhouse/ch"
+	"github.com/uptrace/opentelemetry-go-extra/otelzap"
 	"github.com/uptrace/uptrace/pkg/attrkey"
 	"github.com/uptrace/uptrace/pkg/bunapp"
+	"github.com/uptrace/uptrace/pkg/bunconf"
 	"github.com/uptrace/uptrace/pkg/httperror"
 	"github.com/uptrace/uptrace/pkg/httputil"
 	"github.com/uptrace/uptrace/pkg/org"
-	"go.uber.org/zap"
-	"golang.org/x/exp/maps"
 )
 
+type AlertHandlerParams struct {
+	fx.In
+	Logger *otelzap.Logger
+	Conf   *bunconf.Config
+	PG     *bun.DB
+	CH     *ch.DB
+}
+
 type AlertHandler struct {
-	*bunapp.App
+	*AlertHandlerParams
 }
 
-func NewAlertHandler(app *bunapp.App) *AlertHandler {
+func registerAlertHandler(p bunapp.RouterParams, h *AlertHandler, middleware *Middleware) {
+	p.RouterInternalV1.NewGroup("/projects/:project_id",
+		bunrouter.WithMiddleware(middleware.UserAndProject),
+		bunrouter.WithGroup(func(g *bunrouter.Group) {
+			g.GET("/alerts", h.List)
+			g.GET("/alerts/:alert_id", h.Show)
+			g.PUT("/alerts/closed", h.Close)
+			g.PUT("/alerts/open", h.Open)
+			g.DELETE("/alerts", h.Delete)
+		}))
+}
+
+func NewAlertHandler(p AlertHandlerParams) *AlertHandler {
 	return &AlertHandler{
-		App: app,
+		AlertHandlerParams: &AlertHandlerParams{
```

Comment:
Please use `AlertHandlerParams: p` so we don't have to manually copy each param.